### PR TITLE
Add partial support for non-FHS host systems

### PR DESCRIPTION
**Pull request by roberth**
Fixes weaveworks/weave#2733 partially, where <host>/etc/resolv.conf is not a
symlink

### CODE REVIEW

File: weave [hunk 1]
```diff
@@ -1667,7 +1667,31 @@ launch_router() {
            --label=weavevolumes $WEAVEDB_IMAGE >/dev/null
     fi
 
-    RESOLV_CONF=$(chroot ${HOST_ROOT:-/} readlink -f /etc/resolv.conf)
+    echo "Container PATH is $PATH"
+    # Avoid readlink if not present in FHS directories of host
+    if test -x ${HOST_ROOT:-/}/bin/readlink \
```
Comment (bboreham) [inline weave]: could this be `command_exists readlink` ?

Comment (roberth) [inline weave]: `command_exists` can only check the current environment, not the chroot. I don't know of any standard command that would do it for another root. Something like `chroot ... sh -c "command -v ..."` might work in some cases, even NixOS, but that's executing a binary ("sh") from the host and I'd rather not depend on the host like that.

Comment (rade) [inline weave]: so perhaps simply
1. run `test -L ...` to determine whether we are looking at a symlink
2. if false then no-op
3. if true then attempt the chroot'ed readlink

This doesn't give us a nice error when readlink isn't there but IMO the stock error is quite ok.

Or perhaps even just
```
RESOLV_CONF=/etc/resolv.conf
[ -f ${HOST_ROOT:-/}$RESOLV_CONF ] || RESOLV_CONF=$(chroot ${HOST_ROOT:-/} readlink -f $RESOLV_CONF)
```

**Comment by roberth** [comment]
I've put the symlink condition first

File: weave [hunk 2]
```diff
@@ -1667,7 +1667,14 @@ launch_router() {
            --label=weavevolumes $WEAVEDB_IMAGE >/dev/null
     fi
 
-    RESOLV_CONF=$(chroot ${HOST_ROOT:-/} readlink -f /etc/resolv.conf)
+    if [ -f ${HOST_ROOT:-/}/etc/resolv.conf ]; then # regular file
```
Comment (bboreham) [inline weave]: Sadly `-f` which tests for "regular file" in this context does not exclude symbolic links to files; we need a test such as `-L`

Comment (roberth) [inline weave]: Thanks. I copied that from someone's comment on the previous version of this PR. I will switch back to `-L`

File: weave [hunk 3]
```diff
@@ -1667,7 +1667,16 @@ launch_router() {
...
 
-    RESOLV_CONF=$(chroot ${HOST_ROOT:-/} readlink -f /etc/resolv.conf)
+    # Figure out the location of the actual resolv.conf file because
+    # we want to bind mount its directory into the container.
+    if [ -L ${HOST_ROOT:-/}/etc/resolv.conf ]; then # symlink
+        # This assumes a host with readlink in FHS directories...
+        # Ideally, this would resolve the symlink manually, without
+        # using host commands.
+        RESOLV_CONF=$(chroot ${HOST_ROOT:-/} readlink -f /etc/resolv.conf)
+    else # /etc/resolv.conf is not a symlink
```
Comment (rade) [inline weave]: The comment is misleading since we will also end up in this branch when the file does not exist.

**Comment by bboreham** [comment]
Thanks for the contribution, @roberth !

**Pull request merged**